services/algolia-sync/src/index.js

```diff
@@ -0,0 +1,23 @@
+const minimist = require('minimist');
+const receiveMessage = require('./components/aws-sqs/sqs');
+const siteSync = require('./components/algolia/site-sync');
+
+const { log } = console;
+
+const args = minimist(process.argv.slice(2), {
+  default: {
+    limit: 500,
+    skip: 0,
+  },
+});
+
+const run = () => {
+  if (args.tenant) {
+    siteSync.setup(args.tenant, args.limit, args.skip);
+  } else {
+    log('Waiting for message...');
+    receiveMessage();
```

Comment:
```suggestion
    while (true) {
      process.nextTick(receiveMessage);
      mySynchronousTimingFunction();
    }
```

Ideally this should be implemented by emitting events onto the event loop to take advantage of node's automatic loop tuning, but in the interim this should not be done with recursion. You will eventually hit the maximum call stack size and crash if left as-is.

I would request the ability to set a delay (in ms) between executions to allow on-the-fly tuning.

Reply:
I tried using the while loop. I kept running into issues where the AWS SQS function would loop and not wait for the long pull to finish. Any other ideas on how this could be coded?

Reply:
I updated this